coclico/metrics/metric.py

```diff
@@ -6,19 +6,76 @@
+            computers
+            class_weights (Dict): Dictionnary of the weights of the metric for each class (cf. config file)
+        """
         self.store = store
         self.class_weights = class_weights
 
     def create_metric_intrinsic_jobs(
-        self, name: str, tile_names: List[str], input_path: Path, out_path: Path
+        self, name: str, tile_names: List[str], input_path: Path, out_path: Path, is_ref: bool
     ) -> List[Job]:
-        return [self.create_metric_intrinsic_one_job(name, input_path / f, out_path) for f in tile_names]
+        """Create jobs for a single input classification result folder (eg. ref or c1 or c2 or ...)
+        These jobs are aimed to compute intermediate results relative to the input las only (without comparison) that
+        will be used in `create_metric_relative_to_ref_jobs` to compute the relative metric
+        Args:
+            name (str): classification name (used for job name creation)
+            tile_names (List[str]): list of the filenames of the tiles on which to calculate the result
+            input_path (Path): input folder path (path to the results of the classification)
+            out_path (Path): path for the intermediate results to be saved
+            is_ref (bool): flag that says if the input classification folder is the reference folder (in case it has to
+            be treated differently)
+
+        Returns:
+            List[Job]: List of GPAO jobs to create
+        """
+        return [self.create_metric_intrinsic_one_job(name, input_path / f, out_path, is_ref) for f in tile_names]
+
+    def create_metric_intrinsic_one_job(self, name: str, input: Path, output: Path, is_ref: bool) -> Job:
+        """Create a job for a single input classification result file (one tile).
+        ie. job to create of one element of the `tiles_names` list in create_metric_intrinsic_jobs
 
-    def create_metric_intrinsic_one_job(self, name: str, input: Path, output: Path) -> Job:
+        Args:
+            name (str): classification name (used for job name creation)
+            input (Path): full path of the input tile
+            output (Path): output folder for the result
+            is_ref (bool):  flag that says if the input classification folder is the reference folder (in case it has
+            to be treated differently)
+
+        Raises:
+            NotImplementedError: should be implemented in children classes
+
+        Returns:
+            Job: GPAO job to create
+        """
         raise NotImplementedError
 
     def create_metric_relative_to_ref_jobs(
         self, name: str, out_c1: Path, out_ref: Path, output: Path, c1_jobs: List[Job], ref_jobs: List[Job]
     ) -> List[Job]:
+        """Create jobs to generate a metric that compares the classification results from a c1 folder and th ref folder
+        based on the results of the results of the jobs created by create_metric_intrinsic_jobs
```

Comment:
Pareil, j'aime pas trop result pour parler de l'entrée. Typo: and th ref. Et sur la seconde phrase y'a écrit "the result of the result". Perso j'aurai fait deux phrases, suggestion:
`Create jobs to compute a metric that compares a classification folder (ex: c1) to the reference folder.
It uses intermediary results produced by jobs created by create_metric_intrinsec_jobs`